src/main/java/com/facebook/bitEncoderDecoder/Stage1.java

```diff
@@ -4,6 +4,11 @@
 
     public String encode(String input) {
         StringBuilder encodeString = new StringBuilder();
+        if (input == null) {
+            throw new IllegalArgumentException();
+        } else if (input.equalsIgnoreCase(" ")) {
```

Comment:
na kursie uczyli mnie, żeby equalsa wywoływać nie na dostarczanej zmiennej, a na stałej wartości, bo dzięki temu nie ryzykujemy null pointer exception. tutaj jest to sprawdzane wcześniej, no ale takie bhp :) czyli " ".equals(input)